src/components/Divider.tsx

```diff
@@ -1,50 +1,81 @@
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import rootedParentOf from '../selectors/rootedParentOf'
+import store from '../stores/app'
+import editingValueStore from '../stores/editingValue'
 import fastClick from '../util/fastClick'
+import hashPath from '../util/hashPath'
 import head from '../util/head'
 
+const DIVIDER_PLUS_PX = 20
+const DIVIDER_MIN_WIDTH = 85
+
 /** A custom horizontal rule. */
 const Divider = ({ path }: { path: Path }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
+  const dividerRef = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const [width, setWidth] = useState(DIVIDER_MIN_WIDTH)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
+  /** Get the max width of nearby elements, add DIVIDER_PLUS_PX and set this width for divider. */
   const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.classList.contains('child-divider')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
+    if (dividerRef.current) {
+      const state = store.getState()
+      const parentPath = rootedParentOf(state, path)
+
+      const treeNode = dividerRef.current.closest('div.tree-node') as HTMLElement
+      if (!treeNode) throw new Error('Divider tree node not found')
+
+      const thoughtPath = treeNode.dataset.path
+      if (!thoughtPath || thoughtPath === '') throw new Error('Divider path not found on tree node')
+
+      let elements = document.querySelectorAll(`.tree-node[data-path^="${hashPath(parentPath)}"]:not(.thought-divider)`)
+
+      if (elements.length === 1) {
+        /** If this divider is an only child in a table, find the widest col2. */
+        const grandparentPath = rootedParentOf(state, parentPath)
+        if (!grandparentPath) throw new Error('Divider grandparent not found')
+
+        elements = document.querySelectorAll(
+          `.tree-node.table-col2[data-path^="${hashPath(grandparentPath)}"]:not(.thought-divider)`,
+        )
+      } else if (!elements.length) {
+        /** This is a top-level divider, use its siblings instead. */
+        elements = document.querySelectorAll('.tree-node.top-level:not(.thought-divider)')
+      }
```

Comment:
There's a little more detail [here](https://github.com/cybersemics/em/pull/2108#discussion_r1671843788), but the parentPath of a root/top-level divider is empty, so our query doesn't find any elements to compare - we find top-level nodes instead to calculate width in this case. This will likely change in the rewrite.